crates/weaver_forge/src/filter.rs

```diff
@@ -17,8 +16,9 @@ pub struct Filter {
 impl Filter {
     /// Create a new filter from a string expression or return an error if the
     /// expression is invalid.
-    pub fn try_new(filter_expr: &str) -> Result<Self, Error> {
-        let vars = Vec::new();
+    /// The vars parameter is a list of variable names that can be used in the
+    /// filter expression.
+    pub fn try_new(filter_expr: &str, vars: Vec<String>) -> Result<Self, Error> {
```

Comment:
Nit: Should we pass in a `Map<String,Value>` here for vars?  It feels wierd to have two lists that are dependent on each other in different places.

Reply:
I agree, it's weird but it's the jaq API...